### PR TITLE
👷 Refactor ➾ Added V2 API for the toolkit

### PR DESCRIPTION
**Pull request by mweichert**
# 🌮 Taqueria PR

## 🪁 Description

Previously, the API of the toolkit included two functions that would only work when given Config V1 objects.

This PR adds a V2 namespace for functions that work with Config V2 objects.

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [ ] ⛱️ I have completed this PR template in full and updated the title appropriately
- [ ] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] 🐬 I have commented my code, particularly in hard-to-understand areas
- [ ] 🤿 Corresponding changes have been made to all documentation
- [ ] 🐚 Required changes to the VScE have been made
- [ ] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

- In my last refactor, I introduced a `loadFromEnv` function which has been renamed to `withEnv`, made private, and used internally by two new public functions, `getConfigV1` and `getConfigV2`. These will return a representation of the project's form in either V1 or V2 respectfully, regardless of what version the actual config file is using. This facilitates backwards and frontwards compatibility.
- Introduces the V2 module, which is the start of an API for working with projects using the V2 config. This API is limited to just three functions for now, but we can flesh this out later.

NOTE: this toolkit and the V2 API will be used in the upcoming taco-shop scaffold refactor.

## 🎢 Test Plan

Testing against the taco-shop scaffold.

### CODE REVIEW

File: taqueria-toolkit/TaqError.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export class TaqError extends Error {
+	public isTaqError = true;
+}
+export const isTaqError = (err: unknown): err is TaqError =>
+	typeof err === 'object' && (err as object).hasOwnProperty('isTaqError');
```
Comment (egarson) [inline taqueria-toolkit/TaqError.ts:5]: I would return "isSubclassOf(TaqError)" and you can elide the isTaqError member.
Small fish.

**Review by egarson — APPROVED** [review]
LGTM

File: taqueria-toolkit/TaqError.ts (hunk 1 above)
Comment (mweichert) [reply taqueria-toolkit/TaqError.ts:5]: Hmm, is that new? I'm not aware of TypeScript having this keyword or built-in function. I'd be interested to see how that gets transpiled to JS. We should probably make point of bumping our TS versions as right now we're using 4.7.3.

**Review by jchenche — APPROVED** [review]

File: taqueria-toolkit/TaqError.ts (hunk 1 above)
Comment (egarson) [reply taqueria-toolkit/TaqError.ts:5]: Right, when I said "whatever the equivalent is in TS" I didn't realize the type system has no support for this construct. So never mind :grin:

**Pull request merged**